Sming/Core/Network/TcpConnection.h

```diff
@@ -273,24 +218,22 @@ class TcpConnection
 	uint16_t timeOut = USHRT_MAX; ///< By default a TCP connection does not have a time out
 	bool canSend = true;
 	bool autoSelfDestruct = true;
-#ifdef ENABLE_SSL
-	SSL* ssl = nullptr;
-	SSLCTX* sslContext = nullptr;
-	SSL_EXTENSIONS* sslExtension = nullptr;
+
+	SslContext* sslContext = nullptr;
+	SslConnection* ssl = nullptr;
+	SslExtension* sslExtension = nullptr;
 	bool sslConnected = false;
 	uint32_t sslOptions = 0;
 	SslKeyCertPair sslKeyCert;
 	bool freeKeyCertAfterHandshake = false;
 	SslSessionId* sslSessionId = nullptr;
-#endif
+
 	bool useSsl = false;
```

Comment:
Could wrap all of these up in a single structure or class:

```
struct SslSession {
  Ssl::Context* context = nullptr;
  Ssl::Connection* connection = nullptr;
  Ssl::Extension* extension = nullptr;
  Ssl::SessionId* sessionId = nullptr;

  bool connected = false;
  Ssl::Options options;
  Ssl::KeyCertPair keyCert;
  bool freeKeyCertAfterHandshake = false;

 ~SslSession()
  {
    delete context;
    delete connection;
    delete extension;
  }
};

SslSession* ssl;
```

Don't need `useSsl` any more since we just have a single check for ssl == nullptr. We also get some RAM back for non-SSL use. All (or most) of the related SSL code can also be moved into methods of this class.

Reply:
> Don't need useSsl any more since

useSsl is describing if the user requested SSL. For example when the URL proto is https instead of http.
sslSession is  describing if there is available Ssl implementation.  A tcp client with set ssl can use create SSL or non-SSL connection. If useSsl is true and there is no ssl implementation then the TCP connection should be aborted with also debug error message.